hs_file_types/models/netcdf.py

```diff
@@ -894,3 +787,199 @@ def create_header_info_txt_file(nc_temp_file, nc_file_name):
             dump_file_obj.write("")
 
     return dump_file
+
+
+def netcdf_file_update(instance, nc_res_file, txt_res_file, user):
+    log = logging.getLogger()
+    # check the instance type
+    file_type = isinstance(instance, NetCDFLogicalFile)
+
+    # get the file from irods to temp dir
+    temp_nc_file = utils.get_file_from_irods(nc_res_file)
+    nc_dataset = netCDF4.Dataset(temp_nc_file, 'a')
+
+    try:
+        # update title
+        title = instance.dataset_name if file_type else instance.metadata.title.value
+
+        if title != 'Untitled resource':
+            if hasattr(nc_dataset, 'title'):
+                delattr(nc_dataset, 'title')
+            nc_dataset.title = title
+
+        # update keywords
+        keywords = instance.metadata.keywords if file_type \
+            else [item.value for item in instance.metadata.subjects.all()]
+
+        if hasattr(nc_dataset, 'keywords'):
+            delattr(nc_dataset, 'keywords')
+
+        if keywords:
+            nc_dataset.keywords = ', '.join(keywords)
+
+        # update key/value metadata
+        extra_metadata_dict = instance.metadata.extra_metadata if file_type \
+            else instance.extra_metadata
+
+        if hasattr(nc_dataset, 'hs_extra_metadata'):
+            delattr(nc_dataset, 'hs_extra_metadata')
+
+        if extra_metadata_dict:
+            extra_metadata = []
+            for k, v in extra_metadata_dict.items():
+                extra_metadata.append("{}:{}".format(k, v))
+            nc_dataset.hs_extra_metadata = ', '.join(extra_metadata)
+
+        # update temporal coverage
+        temporal_coverage = instance.metadata.temporal_coverage if file_type \
```

Comment:
I think the NetCDFLogicalFile metadata may not have a temporal coverage in which case it will raise an exception.